cactus/deployment/knownly/engine.py

```diff
@@ -0,0 +1,75 @@
+#coding:utf-8
+import logging
+import os
+
+from cactus.deployment.engine import BaseDeploymentEngine
+from cactus.deployment.knownly.file import KnownlyFile
+from cactus.utils.helpers import get_or_prompt
+from cactus.utils.parallel import multiMap, PARALLEL_DISABLED
+
+logger = logging.getLogger(__name__)
+
+
+class KnownlyDeploymentEngine(BaseDeploymentEngine):
+    FileClass = KnownlyFile
+
+    config_website_directory = 'knownly-website-folder'
+
+    def __init__(self, site):
+        """
+        :param site: An instance of cactus.site.Site
+        """
+        self.site = site
+
+    def deploy(self):
+        self.configure()
+
+        if not os.path.exists(self.target_directory):
+            logger.info("Knownly website folder does not exist. Cancelling " \
+                        "deployment.")
+            return []
+
+        # Upload all files concurrently in a thread pool
+        mapper = multiMap
+        if self.site._parallel <= PARALLEL_DISABLED:
+            mapper = map
```

Comment:
Note to self: need to review whether this will need work with the Python 3 support migration.